documentation/src/test/java/example/exception/ExceptionAssertionExactDemo.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example.exception;
+
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+import org.junit.jupiter.api.Test;
+
+// @formatter:off
+// tag::user_guide[]
+
+public class ExceptionAssertionExactDemo {
+
+    // end::user_guide[]
+    @extensions.ExpectToFail
+    // tag::user_guide[]
+    @Test
+    void testExpectedExceptionIsThrown() {
+        assertThrowsExactly(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Expected exception");
+        }, "This will fail because because we expect IllegalArgumentException to be thrown");
+
+        assertThrowsExactly(RuntimeException.class, () -> {
+            throw new IllegalArgumentException("Expected exception");
+        }, "This will fail because expected exactly RuntimeException to be thrown, not subclasses");
```

Comment:
To the maintainers: Would it make sense to rather have this "This will fail because ..." as comment instead of as message? Because as message it seems a bit weird.

Reply:
Good point, @Marcono1234.

Like with the timeout assertion examples in the [Assertions](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions) section, let's move all "comments" to actual code comments preceding the corresponding assertions, and let's do that consistently across all examples introduced in this PR.

Consequently, let's remove the custom failure messages completely. If we later come up with succinct failure messages that make sense in the examples (which is likely difficult due to the simplicity of the examples), we can add them after this PR, but I'd like to get this PR finished and merged soon (potentially in time for 5.11 M1).